src/Net.jl

```diff
@@ -112,12 +113,51 @@ function chread(
     data
 end
 
+Base.collect(t::Union{Type, DataType, Union{}}) = _collect(t, [])
+_collect(t::Type, list) = t<:Union{} ? push!(list, t) : _collect(t.b, push!(list, t.a))
+_collect(t::Union{DataType,Core.TypeofBottom}, list) = push!(list, t)
+
+function chread(
+    sock::ClickHouseSock,
+    ::Type{Vector{T}},
+    count::VarUInt,
+)::Vector{T} where T <: Union{Missing, Number}
+    type = (T -> for typeName in collect(T)
+                    if typeName !== Missing
+                        return typeName
+                    end
+                end
+    )(T)
+
+    isNull = Vector{Int8}(undef, UInt64(count))
+    data = Vector{type}(undef, UInt64(count))
+    read!(sock.io, isNull)
+    read!(sock.io, data)
+    data = convert(Array{T}, data)
+    map!(data, data, isNull) do x, flag
+        if flag == 1
+            return missing
+        else
+            return x
+        end
+    end
+    return data
+end
```

Comment:
Phew -- took me a few minutes to get what this collect stuff does. We could simplify this quite a lot by just capturing the `T <: Number` type that we need directly in the signature. What do you think about using this code instead:

```suggestion
function chread(
    sock::ClickHouseSock,
    ::Type{Vector{Union{Missing, T}}},
    count::VarUInt,
)::Vector{Union{Missing, T}} where T <: Number
    is_null = Vector{UInt8}(undef, UInt64(count))
    data = Vector{T}(undef, UInt64(count))

    read!(sock.io, is_null)
    read!(sock.io, data)

    collect(map(zip(is_null, data)) do (flag, x)
        flag == 0x1 ? missing : x
    end)
end
```

Additionally, this brings the variable names in line with Julia naming conventions.

In order to keep type support between inserts and selects symmetric, we'd also need a `chwrite` impl for nullable types.